### ISSUE TITLE
The code is inconsistent with the description in the paper for data2vec nlp model ?

### ISSUE THREAD

**Issue by mrhu2017** [facebookresearch/fairseq#4229]
## ❓ Questions and Help

In the description of the data2vec paper,  text data apply the BERT masking strategy,  however, I found that only 80% masked tokens are involved in the loss calculation in the code.

#### Code

self.mask_idx = dictionary.index("<mask>")
.......
masked_indices = src_tokens.eq(self.mask_idx)
x = x[masked_indices]
y = y[masked_indices]
x = self.regression_head(x)
sz = x.size(-1)
if self.cfg.loss_beta == 0:
    loss = F.mse_loss(x.float(), y.float(), reduction="none").sum(dim=-1)
else:
    loss = F.smooth_l1_loss(
                x.float(), y.float(), reduction="none", beta=self.cfg.loss_beta
            ).sum(dim=-1)
